climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+
+        kwargs = dict()
+        kwargs['cmap'] = CMAP_IMPACT
+        kwargs['s'] = 5
+        kwargs['marker'] = ','
+        kwargs['norm'] = BoundaryNorm(
+            np.append(
+                np.append([0], [10 ** x for x in np.arange(0, 2.9, 2.9 / 9)]),
+                [10 ** x for x in np.arange(3, 7, 4 / 90)]),
+            CMAP_IMPACT.N,
+            clip=True
+            )
+
+        # Generate each subplot
+        fig, axis_sub = u_plot.make_map(num_im, proj=proj)
+        if not isinstance(axis_sub, np.ndarray):
+            axis_sub = np.array([[axis_sub]])
+        fig.set_size_inches(9, 8)
+        for array_im, axis, tit, name, coord in zip(list_arr,
+                                                    axis_sub.flatten(),
+                                                    list_tit,
+                                                    list_name,
+                                                    list_coord):
+            if coord.shape[0] != array_im.size:
+                raise ValueError("Size mismatch in input array: %s != %s." % \
+                                 (coord.shape[0], array_im.size))
+            # Binned image with coastlines
+            extent = u_plot._get_borders(coord)
+            axis.set_extent((extent), ccrs.PlateCarree())
+            hex_bin = axis.scatter(coord[:, 1], coord[:, 0], c=array_im, \
+                                   transform=ccrs.PlateCarree(), **kwargs)
+            if shapes:
+                # add warning regions
+                shp = shapereader.Reader(polygon_file)
+                project_crs = lambda x, y: pyproj.transform(pyproj.Proj(init='epsg:21781'),
+                                                            pyproj.Proj(init='epsg:4150'),
+                                                            x, y)
+                for geometry, _ in zip(shp.geometries(), shp.records()):
+                    geom2 = shapely.ops.transform(project_crs, geometry)
+                    axis.add_geometries([geom2],
+                                        crs=ccrs.PlateCarree(),
+                                        facecolor='',
+                                        edgecolor='gray')
+            else: # add country boundaries
+                u_plot.add_shapes(axis)
+            # Create colorbar in this axis
+            cbax = make_axes_locatable(axis).append_axes('bottom',
+                                                         size="6.5%",
+                                                         pad=0.3,
+                                                         axes_class=plt.Axes)
+            cbar = plt.colorbar(hex_bin, cax=cbax, orientation='horizontal',
+                                extend=extend)
+            cbar.set_label(name)
+            cbar.formatter.set_scientific(False)
+            cbar.set_ticks([0, 1000, 10000, 100000, 1000000])
+            cbar.set_ticklabels(['0', "1'000", "10'000", "100'000", "1'000'000"])
```

Comment:
The use of the apostrophe for thousanders is a Swiss uniqueness. I think this should not be the default for a CLIMADA module.

Reply:
I exchanged the ' with spaces